src/chain_access.rs

```diff
@@ -35,146 +45,156 @@ impl LipaChainAccess {
+        self.watched_txs.clear();
+        // Keep just unconfirmed txs to check for confirmations on the next sync.
+        self.watched_txs.extend(unconfirmed_txids);
+        self.watched_txs.extend(pending_txids);
+        // Note: We always keep watched outputs, since LDK does not inform us
+        //       about outputs of interest if they were unspent (unconfirmed).
 
-        Ok(())
+        Ok(SyncResult::Success)
     }
 
-    fn sync_relevant_txs_for_reorgs(&self, confirm: &ConfirmWrapper) -> LipaResult<()> {
-        for txid in confirm.get_relevant_txids().iter() {
-            if !self.esplora.is_tx_confirmed(txid)? {
-                debug!("Transactions reorged out of chain: {:?}", txid);
-                confirm.transaction_unconfirmed(txid);
-            }
+    fn get_block_info(&self, hash: BlockHash) -> LipaResult<Option<BlockInfo>> {
+        if let Some((header, height)) = self.esplora.get_header_with_height(&hash)? {
+            return Ok(Some(BlockInfo {
+                hash,
+                header,
+                height,
+            }));
         }
-
-        Ok(())
+        Ok(None)
     }
 
-    fn inform_about_new_block(
-        &self,
-        confirm: &ConfirmWrapper,
-        cur_tip: &BlockHash,
-    ) -> LipaResult<()> {
-        match self.esplora.get_header_with_height(cur_tip)? {
-            Some((block_header, block_heigh)) => {
-                confirm.best_block_updated(&block_header, block_heigh);
-
-                Ok(())
+    fn filter_unconfirmed(&self, txids: Vec<Txid>) -> LipaResult<Vec<Txid>> {
+        let mut uncofirmed = Vec::new();
+        for txid in txids {
+            if !self.esplora.is_tx_confirmed(&txid)? {
+                uncofirmed.push(txid);
             }
-            None => Err(runtime_error(
-                "Block not found in best chain. Was there a reorg?",
-            )),
         }
+        Ok(uncofirmed)
     }
 
-    #[allow(clippy::result_large_err)]
-    fn sync_txs(&mut self) -> LipaResult<Vec<ConfirmedTransaction>> {
-        let mut confirmed_txs = Vec::new();
-        let mut not_yet_confirmed_txs = HashSet::new();
-
-        debug!("{} transactions to sync", self.watched_txs.len());
-        debug!(
-            "List of transactions to sync: {:?}",
-            self.watched_txs
-                .iter()
-                .map(|tx| tx.0)
-                .collect::<Vec<Txid>>()
-        );
-
-        for tx in self.watched_txs.iter() {
-            let txid = tx.0;
-            if let Some(confirmed_tx) = self.esplora.get_confirmed_tx_by_id(&txid)? {
-                confirmed_txs.push(confirmed_tx);
-                continue;
+    fn split_txs_by_status(
+        &self,
+        txids: HashSet<Txid>,
+    ) -> LipaResult<(Vec<ConfirmedTransaction>, Vec<Txid>)> {
+        let mut confirmed = Vec::new();
+        let mut pending = Vec::new();
+
+        for txid in txids {
+            if let Some(tx) = self.esplora.get_confirmed_tx_by_id(&txid)? {
```

Comment:
Here again, imo an `if let` statement that also has an `else` clause is better off being replaced by a `match` statement.